node/core/provisioner/src/lib.rs

```diff
@@ -385,6 +388,10 @@ async fn collect_disputes(
 				.into_iter()
 				.map(|(s, i, sig)| (DisputeStatement::Invalid(s), i, sig));
 
+			metrics.inc_valid_statements_by(valid_statements.len());
```

Comment:
It's worth noting that the approach of incrementing metrics at this point will count votes many times. Not necessarily a problem, but worth improving documentation about.

What I mean is that the`collect_disputes` function is supposed to return all 'recent' disputes, so that's absolutely every statement from every dispute from the last several sessions - a session being 10 minutes on Rococo, 1 hour on Kusama, and 4 hours on Polkadot. The runtime filters these.

So `collect_disputes` is called every time any node authors a block (I don't believe it should be called if the node isn't authoring a block, but maybe it will) - so it will depend on the number of validators. The metric is probably only useful from a network-wide viewpoint as opposed to a single-node viewpoint, or at least, the values will vary a lot from node to node based on how often they're selected to author blocks.

Reply:
Thanks for the insights, I'll add comments/docs.